core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -970,6 +970,11 @@ protected void runTask(Task task) throws Exception {
             @Override
             protected void onInitMilestoneAttained(InitMilestone milestone) {
                 initLevel = milestone;
+                if (milestone==PLUGINS_PREPARED) {
+                    // set up Guice to enable injection as early as possible
+                    // before this milestone, ExtensionList.ensureLoaded() won't actually try to locate instances
```

Comment:
Which makes me think this whole change is a bad idea, in that it would seem to make fixing [JENKINS-22050](https://issues.jenkins-ci.org/browse/JENKINS-22050) much harder.